### ISSUE TITLE
Login with premium account failing

### ISSUE THREAD

**Issue by askepen** [jpochyla/psst#299]
**Describe the bug**
When logging in today, I get the error: `Authentication failed: premium account required`. 
This was not a problem yesterday and I have checked that my account is still premium on Spotify's website. I have also confirmed  that the password is correct, as i get a different error when typing it wrong deliberately.

**To Reproduce**
Open settings and enter spotify credentials.

**Expected behavior**
No error on login.

**Environment**
Tested on both macOS + ubuntu with latest built action.

**Comment by Anny0nn** [comment]
+1

**Comment by jeromegn** [comment]
Same. I have a Premium Family account.

**Comment by arrudagates** [comment]
Same here, authentication API must have had a change, maybe a fix would be as simple as updating the underlying Spotify libraries being used, unless this has manually implemented the Spotify API.

**Comment by adv4ncr** [comment]
> +1
> 
> edit: (Premium Family account)

same here

**Comment by kingosticks** [comment]
I am seeing this same error on my libspotify-based application, which is not surprising considering they (finally) [killed libspotify today](https://developer.spotify.com/community/news/2022/04/12/libspotify-sunset/).

Is Psst still using something from this older system? I've not got a librespot install on this machine but I don't think anyone has reported issues with that today.

**Comment by arrudagates** [comment]
> I am seeing this same error on my libspotify-based application, which is not surprising considering they (finally) [killed libspotify today](https://developer.spotify.com/community/news/2022/04/12/libspotify-sunset/).
> 
> Is Psst still using something from this older system? I've not got a librespot install on this machine but I don't think anyone has reported issues with that today.

I believe the Psst Spotify API implementation is inspired by librespot, but is completely original and contained within.

**Comment by jpochyla** [comment]
Hi, thanks for the report, I'm on it!

**Comment by kingosticks** [comment]
Yes but there are two versions of the reverse engineered APIs. I thought Psst had implemented the newer version (HTTPS-based CDN, like librespot-Java). Perhaps that doesn't also cover the login system.

**Comment by kingosticks** [comment]
>  I've not got a librespot install on this machine but I don't think anyone has reported issues with that today.

if it's any help, I just found an old librespot install (dev branch from October 2021, not new-api) and it's still working fine.

**Comment by jpochyla** [comment]
librespot is indeed working fine, investigating!

**Comment by fabian-st** [comment]
Installed from Snap, still cant login from spotify premium family ("premium account required")

**Comment by yerke** [comment]
@fabian-st Try to build from source.
